blockchain/chain_sync/src/network_handler.rs

```diff
@@ -46,9 +48,16 @@ impl NetworkHandler {
                         request_id,
                         response,
                     }) => {
-                        rpc_send
-                            .send((request_id, RPCResponse::BlockSync(response)))
-                            .await
+                        let tx = request_table.lock().await.remove(&request_id);
+                        if tx.is_none() {
+                            debug!("RPCResponse receive failed: channel not found");
+                            continue;
+                        }
+                        let tx = tx.unwrap();
+
+                        if let Err(e) = tx.send(RPCResponse::BlockSync(response)) {
+                            debug!("RPCResponse receive failed: {:?}", e)
+                        }
```

Comment:
can do this with an if let or match, seems weird to check none then unwrap? guess that's opinionated tho

Reply:
wow yeah... temporary brain fart here... i hate continues.